### PR TITLE
OTA-1169: Add feature gate for upgrade status

### PR DESCRIPTION
**Pull request by petr-muller**
Add a new `UpgradeStatus` feature gate for the work towards improved upgrade status & health insights information provided via a future `oc adm upgrade status` command.

The code for this feature does not exist yet except for experimental client-side logic implemented in `oc`.

Related: https://issues.redhat.com/browse/OTA-1114
Related: https://github.com/jupierce/oc-update-design-ideas/blob/main/status.md

### CODE REVIEW

File: config/v1/feature_gates.go [hunk 1]
```diff
@@ -431,4 +431,14 @@ var (
 		ResponsiblePerson:   "jhernand",
 		OwningProduct:       ocpSpecific,
 	}
+
+	FeatureGateUpgradeStatus = FeatureGateName("UpgradeStatus")
+	upgradeStatus            = FeatureGateDescription{
+		FeatureGateAttributes: FeatureGateAttributes{
+			Name: FeatureGateUpgradeStatus,
+		},
+		OwningJiraComponent: "over-the-air-updates",
```
[the remark this answers is not in the document]
Comment (petr-muller) [reply config/v1/feature_gates.go:440]: Copied that from existing `SignatureStores` L410, will fix both

**Comment by LalatenduMohanty** [comment]
@deads2k @JoelSpeed Do you need an enhancement PR for this change?

**Comment by JoelSpeed** [comment]
I don't think we generally required an EP for feature gates, and I see a linked design doc in the description anyway. I do think in general though a feature like this (scanning the linked doc) is sufficiently complex that it would benefit from going through the EP process and having feedback from multiple sources

File: config/v1/feature_gates.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (petr-muller) [reply config/v1/feature_gates.go]: Fixed both, let me know if you want me to drop the change for `SignatureStores` as it's technically unrelated to this PR

**Comment by petr-muller** [comment]
Cool! We totally plan to run an EP process round, but having the feature flag would allow us to perform some experimental client-side work that will feed what goes to the EP.

I have fixed up the JIRA component, PTAL

**Comment by deads2k** [comment]
> I don't think we generally required an EP for feature gates, and I see a linked design doc in the description anyway. I do think in general though a feature like this (scanning the linked doc) is sufficiently complex that it would benefit from going through the EP process and having feedback from multiple sources

Agreed.  Recommended, but we have allowed TechPreview featuregates at the team's requests.

/approve

/assign @wking

**Comment by petr-muller** [comment]
Ah they are not optional, just conditionally triggered
/retestr

**Review by LalatenduMohanty — APPROVED** [review]
/lgtm

**Pull request merged**